integration_tests/src/main/python/map_test.py

```diff
@@ -338,6 +338,28 @@ def test_element_at_map_numeric_keys(data_gen):
             'element_at(a, 999)'),
         conf={'spark.sql.ansi.enabled': False})
 
+@pytest.mark.parametrize('data_gen',
+                         [MapGen(StringGen(pattern='key_[0-9]', nullable=False), value(), max_length=6)
+                          for value in get_map_value_gens()],
+                         ids=idfn)
+def test_element_at_map_string_col_keys(data_gen):
+    keys = StringGen(pattern='key_[0-9]')
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: two_col_df(spark, data_gen, keys).selectExpr(
+            'element_at(a, b)', 'a[b]'),
+        conf={'spark.sql.ansi.enabled': False})
+
+@pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
+def test_element_at_map_string_col_keys_ansi_fail(data_gen):
```

Comment:
What about for get map value? Do we need a second test to verify that it works properly in ANSI mode?